src/wallet/scriptpubkeyman.cpp

```diff
@@ -2532,11 +2533,12 @@ SigningResult DescriptorScriptPubKeyMan::SignMessage(const std::string& message,
     return SigningResult::OK;
 }
 
-std::optional<PSBTError> DescriptorScriptPubKeyMan::FillPSBT(PartiallySignedTransaction& psbtx, const PrecomputedTransactionData& txdata, int sighash_type, bool sign, bool bip32derivs, int* n_signed, bool finalize) const
+std::optional<PSBTError> DescriptorScriptPubKeyMan::FillPSBT(PartiallySignedTransaction& psbtx, const PrecomputedTransactionData& txdata, std::optional<int> sighash_type, bool sign, bool bip32derivs, int* n_signed, bool finalize) const
 {
     if (n_signed) {
         *n_signed = 0;
     }
+    if (!sighash_type) sighash_type = SIGHASH_DEFAULT;
```

Comment:
e94bb4e5fc6dea67844a8495b948b679649b7b65: maybe do this inside the `psbtx.tx->vin` loop below and set `SIGHASH_ALL` or `SIGHASH_DEFAULT` depending on whether it's taproot.

And then replace the magic buried in `MutableTransactionSignatureCreator::CreateSig` with an `assert`.

Reply:
This is temporary and will be done inside of `SignPSBTInput` in a later commit.

Reply:
Do you mean in a followup PR? I'm still seeing this in `CreateSig` at the last commit of this PR:

```cpp
// BASE/WITNESS_V0 signatures don't support explicit SIGHASH_DEFAULT, use SIGHASH_ALL instead.
const int hashtype = nHashType == SIGHASH_DEFAULT ? SIGHASH_ALL : nHashType;
```

Reply:
> Is that redundant after [34c33a7](https://github.com/bitcoin/bitcoin/commit/34c33a747128e1c26b1931ac3fa0c4ed6b6a7bb7)?

No. There are other non-PSBT callers of `ProduceSignature` which may still pass `SIGHASH_DEFAULT` to legacy input types, notably `signrawtranssaction`.

Reply:
It might make things easier to follow if those callers pass the right flag as well. If it can be done without making this PR too huge.

Reply:
In other words: for all functions that take a sighash argument, you either pass the correct sighash or nothing.

Reply:
I don't think that's actually correct either as then passing SIGHASH_DEFAULT will be unable to sign for non-taproot at the same time.

It's also a much larger change that I think is out of scope for this PR.

Reply:
> at the same time

You mean a transaction with both a taproot and non-taproot input?

Reply:
> You mean a transaction with both a taproot and non-taproot input?

Yes